apps/src/dance/DanceAiModal.tsx

```diff
@@ -144,25 +191,122 @@ const DanceAiModal: React.FunctionComponent<DanceAiProps> = ({onClose}) => {
       response
     );
     const pickedResponseJson = JSON.stringify(pickedResponse);
-    const formattedPickedResponseJson = pickedResponseJson
-      .replace(/":/g, '": ')
-      .replace(/","/g, '", "')
-      .replace(/,"/g, ', "');
 
     // The block value will be set to this JSON.
     setResponseJson(pickedResponseJson);
 
-    // The user will see this function call.
-    setResponseCall(`ai(${formattedPickedResponseJson})`);
-
     // The user will see this explanation.
     setResponseExplanation(response.explanation);
   };
 
+  const convertBlocks = () => {
+    const params = JSON.parse(responseJson);
+
+    const blocks = [
+      Blockly.getMainWorkspace().newBlock('Dancelab_setForegroundEffect'),
+      Blockly.getMainWorkspace().newBlock(
+        'Dancelab_setBackgroundEffectWithPalette'
+      ),
+      Blockly.getMainWorkspace().newBlock('Dancelab_makeNewDanceSpriteGroup'),
+    ];
+
+    // Foreground block.
+    blocks[0].setFieldValue(params.foregroundEffect, 'EFFECT');
+
+    // Background block.
+    blocks[1].setFieldValue(params.backgroundEffect, 'EFFECT');
+    blocks[1].setFieldValue(params.backgroundColor, 'PALETTE');
+
+    // Dancers block.
+    blocks[2].setFieldValue(params.dancers.type.toUpperCase(), 'COSTUME');
+    blocks[2].setFieldValue(params.dancers.count.toString(), 'N');
+    blocks[2].setFieldValue(params.dancers.layout, 'LAYOUT');
+
+    const origBlock = currentAiModalField?.getSourceBlock();
+
+    if (
+      origBlock &&
+      currentAiModalField &&
+      blocks[0] &&
+      blocks[1] &&
+      blocks[2] &&
+      blocks[0].previousConnection &&
+      blocks[0].nextConnection &&
+      blocks[1].previousConnection &&
+      blocks[1].nextConnection &&
+      blocks[2].previousConnection &&
+      blocks[2].nextConnection
+    ) {
+      blocks[0].nextConnection.connect(blocks[1].previousConnection);
+      blocks[1].nextConnection.connect(blocks[2].previousConnection);
+
+      const blocksSvg = blocks.map(block => block as BlockSvg);
```

Comment:
nit: I think this can just be `const blocksSvg = blocks as BlockSvg[]` unless TS claims there's no overlap?

Reply:
Good call.  As discussed with you further, we can just use `BlockSvg` for everything here, since it extends `Block`.